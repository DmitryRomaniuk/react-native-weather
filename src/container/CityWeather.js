import React, {Component} from "react";
import {
    StyleSheet,
    View,
    Dimensions,
    StatusBar
} from "react-native";
import {
    Container,
    Content,
    Header,
    Grid,
    Col,
    Row,
    Title,
    Body,
    Item,
    Left,
    Right,
    Input,
    Icon,
    Button,
    Text
} from 'native-base';

let {width} = Dimensions.get("window");

const styles = StyleSheet.create({
    container: {
        flex: 1,
        // paddingTop: 40,
        alignItems: "center",
        justifyContent: "flex-start"
    },
    firstColumn: {
        alignItems: 'flex-start',
        justifyContent: "flex-start",
        width: 0.4 * width,
    },
    secondColumn: {
        alignItems: 'center',
        justifyContent: "center",
    },
    textColumn: {
        fontSize: 24,
    },
});

export default class CityWeather extends Component {
    constructor(props) {
        super(props);
        this.weather = this.props.navigation.state.params.city;
        this.state = {
            text: "",
        };
    }

    static navigationOptions = {
        header: null,
    };

    componentWillMount() {
        StatusBar.setHidden(true);
    }

    goBack = () => {
        this.props.navigation.goBack();
    };

    render() {
        return (
            <Container>
                <Header searchBar rounded>
                    <Left>
                        <Button transparent onPress={this.goBack}>
                            <Icon name='arrow-back'/>
                        </Button>
                    </Left>
                    <Body>
                    <Title>Weather {this.props.navigation.state.params.city.name}</Title>
                    </Body>
                </Header>
                <Content>
                    <Grid>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>Country</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text style={styles.textColumn}>{this.weather.sys.country}</Text>
                            </Col>
                        </Row>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>City</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text style={styles.textColumn}>{this.weather.name}</Text>
                            </Col>
                        </Row>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>{this.weather.weather[0].main}</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text style={styles.textColumn}>{this.weather.weather[0].description}</Text>
                            </Col>
                        </Row>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>Temperature</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text
                                    style={styles.textColumn}>{(this.weather.main.temp > 150) ? this.weather.main.temp - 273.15 : this.weather.main.temp} &#x2103;</Text>
                            </Col>
                        </Row>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>Pressure</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text
                                    style={styles.textColumn}>{Math.round(this.weather.main.pressure * 100 * 760 / 101325)}
                                    mmHg</Text>
                            </Col>
                        </Row>
                        <Row>
                            <Col style={styles.firstColumn}>
                                <Text style={styles.textColumn}>Wind</Text>
                            </Col>
                            <Col style={styles.secondColumn}>
                                <Text style={styles.textColumn}>{this.weather.wind.speed} m/s</Text>
                            </Col>
                        </Row>
                    </Grid>
                </Content>
            </Container>
        );
    }
}