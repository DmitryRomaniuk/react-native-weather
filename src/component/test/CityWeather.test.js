import React from 'react';
import CityWeather from '../CityWeather';
// import Enzyme from 'enzyme';
// import { shallow, mount, render } from 'enzyme';
// import sinon from 'sinon';
// import Adapter from 'enzyme-adapter-react-16';
// Enzyme.configure({ adapter: new Adapter() });

import renderer from 'react-test-renderer';

describe('Test CityWeather component', () => {
    let navigation;

    beforeEach(() => {
        navigation = {
            state: {
                params: {
                    city: {
                        name: '',
                        sys: {country: ''},
                        weather: [{main: '', description: ''}],
                        main: {temp: 286.15, pressure: 101325},
                        wind: {speed: ''},
                    }
                }
            }
        };
    });

    test('renders without crashing', () => {
        const rendered = renderer.create(<CityWeather navigation={navigation}/>).toJSON();
        expect(rendered).toBeTruthy();
    });

    test('renders correctly', () => {
        const tree = renderer.create(
            <CityWeather navigation={navigation}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it('should call goBack', () => {
    //     let component = shallow(<CityWeather navigation={navigation}/>);
    //     const instance = component.instance();
    //     const handleClickSpy = sinon.spy(instance, 'goBack');
    //     component.find('button').simulate('click');
    //     expect(handleClickSpy.called).to.equal(true);
    // });
});