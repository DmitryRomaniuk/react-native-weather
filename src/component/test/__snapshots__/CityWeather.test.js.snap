// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CityWeather component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      rounded={true}
      searchBar={true}
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 33,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 2,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "ExponentFont-mock-ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            Ôèè
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Weather 
          
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Country
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              City
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Temperature
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              13 ‚ÑÉ
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Pressure
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              76000mmHg
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 0,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": 300,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Wind
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
               m/s
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
